funding/commitment_type_negotiation.go

```diff
@@ -223,6 +223,51 @@ func explicitNegotiateCommitmentType(channelType lnwire.ChannelType, local,
 		}
 		return lnwallet.CommitmentTypeTweakless, nil
 
+	// Simple taproot channels only.
+	case channelFeatures.OnlyContains(lnwire.SimpleTaprootChannelsRequired):
+
+		if !hasFeatures(
+			local, remote, lnwire.SimpleTaprootChannelsOptional,
+		) {
+			return 0, errUnsupportedChannelType
+		}
+
+		return lnwallet.CommitmentTypeSimpleTaproot, nil
+
+	// Simple taproot channels with scid only.
+	case channelFeatures.OnlyContains(
+		lnwire.SimpleTaprootChannelsRequired,
+		lnwire.ScidAliasRequired,
+	):
+
+		if !hasFeatures(
+			local, remote,
+			lnwire.SimpleTaprootChannelsOptional,
+			lnwire.ScidAliasOptional,
+		) {
+			return 0, errUnsupportedChannelType
+		}
+
+		return lnwallet.CommitmentTypeSimpleTaproot, nil
+
+	// Simple taproot channels with zero conf only.
+	case channelFeatures.OnlyContains(
+		lnwire.SimpleTaprootChannelsRequired,
+		lnwire.ScidAliasRequired,
+		lnwire.ZeroConfRequired,
+	):
+
+		if !hasFeatures(
+			local, remote,
+			lnwire.SimpleTaprootChannelsOptional,
+			lnwire.ScidAliasOptional,
+			lnwire.ZeroConfOptional,
```

Comment:
not sure how important this is but: for the other cases where we check `OnlyContains` for both `ScidAlias` and `ZeroConf`, we then only check `hasFeatures` for `ZeroConf` since `ScidAlias` can be implied from `ZeroConf` and so I dont think it explicitly needs to be set in the feature set.

Reply:
good catch, I think it's not meant to include both at once in the channel type. Looking back at the zero-conf spec PR my question about this was never answered, so I think we should just keep the status quo here and go with your suggestion